### DISCUSSION TITLE
LightCompnent in submarine file, in bot's inventory, in player's inventory

### DISCUSSION THREAD

**Discussion by IrLED** [FakeFishGames/Barotrauma#15220] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Light component is properly loaded/initialized when it's the part of submarine file.
When it is the part of bot inventory (gamesession.xml) - it loads only on/off state, color/pulsing/flickering is default.
When the part of player inventory (_CharacterData.xml)- component is loaded as off, color/pulsing/flickering is default.

### Reproduction steps

1. In sub editor add a 2 light components to the submarine (e.g. dugong) with not default flickering/pulsing/color
2. create a campaign with that modified sub
3. observe lights on the wall functioning properly
4. Hire a bot from HR Office (not console spawn)
5. detach lights from the wall, give one to the bot(equipped), the other leave in your inventory(equipped)
6. [Save and Quit]
7. Observe proper values in saved gamesession.xml, _CharacterData.xml
8. load the save
9. Observe bot's light on (default white, no pulsing), player's light off (on dedicated, not always reproducible)/on (on in-game), color/pulsing is default
10. collect components and attach to the wall of the sub
11. [Save and Quit]
12. load the save
13. observe lights on the wall functioning properly

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Other

### -

dedicated and in-game

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! This has been addressed in the current unstable build (PR in our private repo https://github.com/FakeFishGames/Barotrauma-development/pull/5724), fix coming up in the next update.
